internal/resources/fetcher_test.go

```diff
@@ -284,3 +288,203 @@ func TestGetServiceByAppLabel(t *testing.T) {
+		"kind":       "Namespace",
+		"metadata": map[string]interface{}{
+			"name":              "testingns",
+			"creationTimestamp": nil,
+		},
+		"spec":   map[string]interface{}{},
+		"status": map[string]interface{}{},
+	}
+
+	expectedResource := []AuditableResources{{
+		Policies:  []policiesv1.Policy{&policy},
+		Resources: []unstructured.Unstructured{{Object: unstructuredNamespace}},
+	}}
+
+	apiResourceList := metav1.APIResourceList{
+		GroupVersion: "v1",
+		APIResources: []metav1.APIResource{
+			{
+				Name:         "namespaces",
+				SingularName: "namespace",
+				Kind:         "Namespace",
+				Namespaced:   false,
+			},
+			{
+				Name:         "pods",
+				SingularName: "pod",
+				Kind:         "Pod",
+				Namespaced:   true,
+			},
+		},
+	}
+
+	fakeClientSet := fakekubernetes.NewSimpleClientset()
+	fakeClientSet.Resources = []*metav1.APIResourceList{&apiResourceList}
+
+	fetcher := Fetcher{dynamicClient, "", "", fakeClientSet}
+
+	resources, err := fetcher.GetClusterWideResourcesForPolicies(context.Background(), []policiesv1.Policy{&policy})
+	if err != nil {
+		t.Errorf("unexpected error: " + err.Error())
+	}
+	if !cmp.Equal(resources, expectedResource) {
+		diff := cmp.Diff(expectedResource, resources)
+		t.Errorf("Expected AuditableResources differs from the expected value: %s", diff)
+	}
+}
+
+func TestIsNamespacedResource(t *testing.T) {
+	tests := []struct {
+		name                 string
+		apiResourceList      metav1.APIResourceList
+		gvr                  schema.GroupVersionResource
+		expectedIsNamespaced bool
+		expectedErr          error
+	}{
+		{"pods", metav1.APIResourceList{
+			GroupVersion: "v1",
+			APIResources: []metav1.APIResource{
+				{
+					Name:         "namespaces",
+					SingularName: "namespace",
+					Kind:         "Namespace",
+					Namespaced:   false,
+				},
+				{
+					Name:         "pods",
+					SingularName: "pod",
+					Kind:         "Pod",
+					Namespaced:   true,
+				},
+			},
+		},
+			schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "pods",
+			}, true, nil,
+		},
+		{"namespaces", metav1.APIResourceList{
+			GroupVersion: "v1",
+			APIResources: []metav1.APIResource{
+				{
+					Name:         "namespaces",
+					SingularName: "namespace",
+					Kind:         "Namespace",
+					Namespaced:   false,
+				},
+				{
+					Name:         "pods",
+					SingularName: "pod",
+					Kind:         "Pod",
+					Namespaced:   true,
+				},
+			},
+		},
+			schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "namespaces",
+			}, false, nil,
+		},
+		{"resource not found", metav1.APIResourceList{
+			GroupVersion: "v1",
+			APIResources: []metav1.APIResource{
+				{
+					Name:         "namespaces",
+					SingularName: "namespace",
+					Kind:         "Namespace",
+					Namespaced:   false,
+				},
+				{
+					Name:         "pods",
+					SingularName: "pod",
+					Kind:         "Pod",
+					Namespaced:   true,
+				},
+			},
```

Comment:
Shouldn't this be an empty list?

Reply:
This is the  list returned by the k8s api mock. Thus, the audit scanner should validate that resources returned by the API contains or not the resource requested. You can notice that in the `expectedIsNamespaced` and `expectedErr` fields. Where the expected values are `false` and `constants.ErrResourceNotFound` respectively.